### PR TITLE
Changes on bootloader to support HPUX, with python 2.7 and gcc.

### PR DESCRIPTION
**Pull request by goodtiding5**
Add support for HPUX for python 2.7 and gcc compiler.

### CODE REVIEW

**Comment by htgoebel** [comment]
Thanks for this enhancement :-) Unfortunately the tests fail:
```
  File "/home/travis/build/pyinstaller/pyinstaller/bootloader/wscript", line 466
    ctx.env.append_value('DEFINES', 'HPUX')
                                          ^
TabError: inconsistent use of tabs and spaces in indentation
```

Please fix and update the pull-request. I also added some line-comments. We'd appreciate if you'd fix the existing commits (e.g. `git rebase -i ...`) instead of adding new ones, as the formet would make reading the change easier in a few years time. Thanks.

For updating a pull-request, you can simply (force) push the updated branch to github again. This will automatically update the pull-request (which follows the branch, not the commit). So you do not need to close the pull-request and open a new one. This also has the benefit that the discussion history is kept.

File: PyInstaller/compat.py [hunk 1]
```diff
@@ -74,6 +75,9 @@
                      'libpython%d.%dm.so.1' % _pyver,
                      'libpython%d.%d.so.1.0' % _pyver,
                      'libpython%d.%dm.so.1.0' % _pyver}
+elif is_hpux:
+    PYDYLIB_NAMES = {'libpython%d.%d.a' % _pyver,
```
Comment (htgoebel) [inline PyInstaller/compat.py]: I assume the `.a` is wrong and should be removed.

`.a` is a static library on most Unix systems. Is this different at HP-UX?

File: bootloader/wscript [hunk 2]
```diff
@@ -614,7 +642,7 @@ def build(ctx):
         elif is_aix:
             libs = ['dl', 'm']
             staticlibs = ['z']
-
+
```
Comment (htgoebel) [inline bootloader/wscript]: Useless change of whitespace (and trailing whitespace, too).

Comment (goodtiding5) [inline bootloader/wscript]: Agreed!

File: PyInstaller/compat.py (hunk 1 above)
Comment (goodtiding5) [inline PyInstaller/compat.py]: Let me verify that.

**Comment by goodtiding5** [comment]
OK. I made changes according to your suggestion.  I didn't know that TAB is now allowed here :-)

File: PyInstaller/compat.py (hunk 1 above)
Comment (goodtiding5) [inline PyInstaller/compat.py]: You are right!  I made the change accordingly.

**Pull request merged**